invoice/src/main/java/org/killbill/billing/invoice/api/user/DefaultInvoiceUserApi.java

```diff
@@ -269,23 +269,32 @@ public Invoice getInvoiceByInvoiceItem(final UUID invoiceItemId, final TenantCon
     public Invoice triggerDryRunInvoiceGeneration(final UUID accountId, final LocalDate targetDate, final DryRunArguments dryRunArguments, final CallContext context) throws InvoiceApiException {
         final InternalCallContext internalContext = internalCallContextFactory.createInternalCallContext(accountId, context);
 
-        final Invoice result = dispatcher.processAccount(true, accountId, targetDate, dryRunArguments, false, internalContext);
-        if (result == null) {
+        final Iterable<Invoice> results = dispatcher.processAccount(true, accountId, targetDate, dryRunArguments, false, internalContext);
+        if (results == null) {
             throw new InvoiceApiException(ErrorCode.INVOICE_NOTHING_TO_DO, accountId, targetDate != null ? targetDate : "null");
         } else {
-            return result;
+            if (Iterables.<Invoice>size(results) != 1) {
+                // If we change the API one day, we need to fix the implementation (see comments in processAccountWithLockAndInputTargetDate)
+                throw new UnsupportedOperationException("Invoices were generated but this version of Kill Bill doesn't support returning grouped invoices");
```

Comment:
Could we instead return one Invoice with all item (non grouped). The `invoiceId` that is returned in dry-run is bogus anyways. We would lose the grouping but still have the correct set of items

Reply:
I find it confusing: dry-run wouldn't reflect the non-dry-run case (grouping).